src/utils/pagerduty/resolve-incidents.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-console */
+
+import {sleep} from '../sleep'
+
+export const resolveIncidents = async (
+  pdClient: any,
+  serviceId: string,
+  testService: string,
+) => {
+  const allIncidents = await pdClient.all(
+    `/incidents?service_ids%5B%5D=${serviceId}&statuses%5B%5D=acknowledged&statuses%5B%5D=triggered`,
+  )
+  const allOpenIncidents = allIncidents.data
+  .reduce((acc: any, i: any) => {
+    if (i.incidents !== undefined && i.incidents.length > 0) {
+      acc = [...acc, ...i.incidents]
+    }
+    return acc
+  }, [])
+  .filter((i: any) => i.title.split(' - Tests: ')[0] === testService)
```

Comment:
Is there any way at indecent creation to get the same incident (like the way once an incident in acknowledged is not triggered again) then to store on each of them the source ID of the first incident? 
This to avoid the string check on the title, ease the filtering (close all incidents with the same parent ID)

Reply:
I checked in the pagerduty API, and it is not possible to add such metadata to incidents (such as a tag), see discussion here: https://community.pagerduty.com/forum/t/tag-or-label-to-categorize-incidents/974. I verified in the API specs and did not find anything either in the endpoints for creation and list (https://developer.pagerduty.com/api-reference/9d0b4b12e36f9-list-incidents)

Using the dedup key (incident_key) wouldn't help us since we already have a mechanism to not create multiple incidents with the same dedup key. In our situation, we want to resolve past incidents with a different dedup key.

It seems only possible to do that via parsing either content in the title or content in the description of the incident.